app/posts/[url]/page.tsx

```diff
@@ -0,0 +1,34 @@
+import { allPosts } from '@/.contentlayer/generated';
+
+interface PageProps {
+  params: {
+    url: string;
+  };
+}
+
+export async function generateStaticParams(): Promise<PageProps['params'][]> {
+  return allPosts.map(({ url }) => ({
+    url: url,
+  }));
+}
+
+export default function PostPage({ params }: PageProps) {
+  const post = allPosts.find(
+    ({ url }) => url.replaceAll('/posts/', '') === params.url,
+  );
+
+  if (!post) {
+    return null;
```

Comment:
I think it would be better to use [notFound](https://nextjs.org/docs/app/api-reference/functions/not-found) here to redirect to the not found page instead of showing a blank page